pkg/querier/blocks_finder_bucket_index.go

```diff
@@ -112,3 +114,7 @@ func (f *BucketIndexBlocksFinder) GetBlocks(ctx context.Context, userID string,
 
 	return blocks, matchingDeletionMarks, nil
 }
+
+func newBucketIndexTooOldError(updatedAt time.Time, maxStalePeriod time.Duration) error {
+	return errors.New(globalerror.BucketIndexTooOld.MessageWithLimitConfig(fmt.Sprintf("bucket index is too old. It was last updated at %s which exceeds the maximum allowed staleness period of %v", updatedAt.UTC().Format(time.RFC3339Nano), maxStalePeriod), tsdb.BucketIndexConfigPrefix+tsdb.MaxStalePeriodFlag))
```

Comment:
Nit before applying this: golang errors have to start with lowercase.

If we're going to hack that rule (which may make sense since we're building end-user facing messages here, not just internal wrappable errors), we should make the hacking standard.

Reply:
Applied accept for the initial capital. If we decide to change that, let's do it in a separate PR since it would affect many errors.

Reply:
Thank you, @colega. Consistency wins in most cases. Thank you for the pushback. Good call, @zenador.

Reply:
No separate PR needed. Let’s keep this as is per @colega’s rationale.